xbmc/pvr/recordings/PVRRecording.cpp

```diff
@@ -554,8 +555,17 @@ bool CPVRRecording::IsInProgress() const
   // Note: It is not enough to only check recording time and duration against 'now'.
   //       Only the state of the related timer is a safe indicator that the backend
   //       actually is recording this.
-
-  return GetRecordingTimer() != nullptr;
+  // Once the recording is known to not be in progress that will never change.
+  if (m_bMaybeInProgress)
+  {
+    bool reallyInProgress = GetRecordingTimer() != nullptr;
+    m_bMaybeInProgress = reallyInProgress;
+    return reallyInProgress;
```

Comment:
no need for the extra variable here (and rename the member var, see below).

```c++
// Once the recording is known to not be in progress that will never change.
if (m_bInProgress)
  m_bInProgress = GetRecordingTimer() != nullptr;

return m_bInProgress;
```

Reply:
Done, pushed, squashed